pkg/controller/core/localqueue_controller.go

```diff
@@ -183,6 +183,10 @@ func (r *LocalQueueReconciler) Update(e event.UpdateEvent) bool {
 	log := r.log.WithValues("localQueue", klog.KObj(newLq))
 	log.V(2).Info("Queue update event")
 
+	if features.Enabled(features.LocalQueueMetrics) {
+		updateLocalQueueResourceMetrics(newLq)
+	}
+
```

Comment:
This is a good catch. Thanks!

I think we also need to cherry pick it somehow. @mimowo WDYT, is it ok to keep it here? Or should we create prepare PR just for this fix?

Reply:
> What is this code change fixing? Do we have a test demonstrating the fix?

The `updateLocalQueueResourceMetrics(newLq)` never run. We should execute this statement before all `if` statefments.

> Yes, Ideally we can isolate the fix and cherry pick.

Could you please prepare separate PR for this fix? Maybe we also have time to add it to v0.10.1.

Reply:
> What is this code change fixing? Do we have a test demonstrating the fix?

The `updateLocalQueueResourceMetrics(newLq)` never run. We should execute this statement before all `if` statefments.

> Yes, Ideally we can isolate the fix and cherry pick.

@mykysha Could you please prepare separate PR for this fix? Maybe we also have time to add it to v0.10.1.

Reply:
Merged in the https://github.com/kubernetes-sigs/kueue/pull/3988